packages/core/src/lib/connection_manager.ts

```diff
@@ -0,0 +1,381 @@
+import type { Connection } from "@libp2p/interface-connection";
+import type { PeerId } from "@libp2p/interface-peer-id";
+import type { PeerInfo } from "@libp2p/interface-peer-info";
+import type { Peer } from "@libp2p/interface-peer-store";
+import type { IRelay } from "@waku/interfaces";
+import { Tags } from "@waku/interfaces";
+import debug from "debug";
+import type { Libp2p } from "libp2p";
+
+import { KeepAliveManager, KeepAliveOptions } from "./keep_alive_manager.js";
+
+export const DEFAULT_MAX_BOOTSTRAP_PEERS_ALLOWED = 1;
+// TODO: add this functionality back
+// const DEFAULT_APPROX_TIME_BEFORE_BOOTSTRAP_FALLBACK_MS = 5000;
+export const DEFAULT_MAX_DIAL_ATTEMPTS_FOR_PEER = 3;
+export const DEFAULT_LOGGING_INTERVAL_MS = 10_000;
+
+const log = debug("waku:connection-manager");
+
+export interface Libp2pComponents {
+  connectionManager: Libp2p["connectionManager"];
+  peerStore: Libp2p["peerStore"];
+  ping: Libp2p["ping"];
+  dial: Libp2p["dial"];
+}
+
+export interface UpdatedStates {
+  allPeers: Peer[];
+  allDialablePeers: Peer[];
+  bootstrapPeers: Peer[];
+  nonBootstrapPeers: Peer[];
+  dialableBootstrapPeers: Peer[];
+  dialableNonBootstrapPeers: Peer[];
+  allConnections: Connection[];
+  bootstrapConnections: Connection[];
+  nonBootstrapConnections: Connection[];
+}
+
+export interface Options {
```

Comment:
Is it intended to be used outside of the package? If so I would rename it to something like `ConnectionManagerOption` or remove export of it.

Reply:
I see, maybe let's rename it to be more precise?

Reply:
If you agree then let's include more verbose naming already here :) and follow up on it in other places in this issue https://github.com/waku-org/js-waku/issues/1147